tools/c7n_azure/c7n_azure/resources/network_security_group.py

```diff
@@ -52,223 +53,204 @@ class SecurityRuleFilter(Filter):
-                        return False
-                else:
-                    if dest_port_range[0] == port:
-                        return False
-        # Ports is specified, only those included in Ports should return true
-        elif self.ports:
-            if len(dest_port_range) > 1:
-                # self.ports needs to have ALL ports in range (inclusive) to match
-                range_set = set(range(dest_port_range[0], dest_port_range[1] + 1))
-                ports_set = set(self.ports)
-                return range_set.issubset(ports_set)
-            else:
-                return dest_port_range[0] in self.ports
-        return True
+    def check_nsg(self, nsg):
+        nsg_ports = PortsRangeHelper.build_ports_array(nsg, self.direction_key, self.ip_protocol)
 
-    """
-    Check to see if port ranges defined in security rule match range as defined by policy
-    """
-    def is_ranges_match(self, security_rule):
-        if not any([self.from_port, self.to_port, self.except_ports, self.ports]):
-            return True
-        if 'destinationPortRange' in security_rule['properties']:
-            dest_port_ranges = \
-                [self.get_port_range(security_rule['properties']['destinationPortRange'])]
-        else:
-            dest_port_ranges = \
-                [self.get_port_range(range_str) for range_str
-                 in security_rule['properties']['destinationPortRanges']]
-        for range in dest_port_ranges:
-            if not self.is_range_match(range):
-                return False
-        return True
-
-    def get_port_range(self, range_str):
-        return [int(item) for item in range_str.split('-')]
+        allow = len([p for p in self.ports if nsg_ports[p]])
+        deny = len(self.ports) - allow
 
-    """
-    Determine if SecurityRule matches criteria as entered in policy
-
-    Currently supporting filters:
-        {} - Specific Ports to target
-        {} - Ports to IGNORE
-        {} - Lower bound of port range (inclusive)
-        {} - Upper bound of port range (inclusive)
-        {} - TCP/UDP protocol
-        {} - Allow/Deny access
-    """.format(PORTS, EXCEPT_PORTS, FROM_PORT, TO_PORT, IP_PROTOCOL, ACCESS)
-
-    def is_match(self, security_rule):
-        if not StringUtils.equal(self.direction_key, security_rule['properties']['direction']):
-            return False
-        ranges_match = self.is_ranges_match(security_rule)
-        protocol_match = (self.ip_protocol is None) or \
-                         (StringUtils.equal(self.ip_protocol,
-                                            security_rule['properties']['protocol']))
-
-        if self.access is not None:
-            access_match = StringUtils.equal(self.access, security_rule['properties']['access'])
-            return self.match_op([ranges_match, protocol_match, access_match])
-        else:
-            return self.match_op([ranges_match, protocol_match])
+        if self.ports_op == 'all':
```

Comment:
Add documentation for ports-op. It was difficult for me to tell what the user's options were and what they did until scrolling finally down to here and examining the code.